torchtext/data/iterator.py

```diff
@@ -65,6 +66,10 @@ def __init__(self, dataset, batch_size, sort_key=None, device=None,
                            + " or passing a string as an argument. This behavior will be"
                            + " deprecated soon and currently defaults to cpu.")
             device = None
+
+        if device is None:
+            device = torch.device('cpu')
```

Comment:
In the doc, device could also be string. Should we also cover that as:
```
if device is None:
    device = torch.device('cpu')
elif isinstance(device, str):
    device = torch.device(device)
```

Reply:
I would also like to add one more thing. How about adding a warning if cuda is available and user still preferred 'cpu'
```
if torch.cuda.is_available():
    if device is None or device=='cpu':
        logger.warning("you have gpu available but preferred device as cpu")
if device is None:
    device = torch.device('cpu')
elif isinstance(device, str):
    device = torch.device(device)
```

Reply:
I wouldn't assume that GPU is preferred if it is available. This might be used in some unexpected context. 

Alongside updating the default behavior we should also update the doc string to document this.

Reply:
Sometimes users forget to pass 'gpu' parameter even if gpu is available. So, if we send a warning, users can rectify it.

Reply:
Yes, but it adds verbosity we commonly don't have. I suggest we do this in a separate PR where we address logging issues in general. If we were to do it for this operator we should also do it for the other ones and that'll required some strategy.

Reply:
ok then, for now i will ignore and add only required ones.